### PR TITLE
docs(adr): add apache-2.0 license ADR

### PR DESCRIPTION
**Pull request by daveworth**
This PR introduces an ADR, as proposed by Jeff and strongly supported by myself to ensure that all UDS Products are under the Apache-2.0 software license.  It also introduces an ADR directory within `docs/` (though I am open to moving to the project root).

### CODE REVIEW

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## Status
+
+Accepted
+
+## Context
+
+Product has been considering a copy-left (LGPL) license for some time for all
+UDS Products. This was a consideration in light of the behavior of several
+business lately moving towards source-available licenses after gaining scale for
```
Comment (zachariahmiller) [inline docs/adrs/0002-apache-2.0-for-all-uds-products.md:13]: businesses

**Review by zachariahmiller — COMMENTED** [review]
@daveworth left a couple comments/nits. Also i think since this is an ADR regarding licensing and doesnt impacts the functionality of the code in this repo the semantic commit could be something like `docs(adr)` instead of `feat(adr)` as to not trigger a minor release of the code base.

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md (hunk 1 above)
Comment (daveworth) [reply docs/adrs/0002-apache-2.0-for-all-uds-products.md:13]:
```suggestion
businesses lately moving towards source-available licenses after gaining scale for
```

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Apache-2.0 to avoid hindering adoption by our competitors. I believe this aligns
+very well with our culture and is the kind of company I want to be a part of.
+
+## Decision
+
+As a business we have decided to keep Apache-2.0 for all of our UDS products.
+
+## Consequences
+
+Some of our competitors may take UDS, slap their logo on it, and sell
```
[the remark this answers is not in the document]
Comment (daveworth) [reply docs/adrs/0002-apache-2.0-for-all-uds-products.md:45]:
```suggestion
Could some of our competitors take UDS, slap their logo on it, and sell
```

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+it/underbid us in some cases? Absolutely. We acknowledge that is possible.  But
```
[the remark this answers is not in the document]
Comment (daveworth) [reply docs/adrs/0002-apache-2.0-for-all-uds-products.md:46]:
```suggestion
it and/or underbid us in some cases? Absolutely. We acknowledge that is possible.  But
```

**Review by zachariahmiller — DISMISSED** [review]
LGTM barring any additional changes @jeff-mccoy or @Racer159 might suggest.

**Comment by daveworth** [comment]
@Racer159 *thank you* for the updated title.  I was just noting yesterday that I don't like `chore` as the conventional commit message around docs and @Noxsios reminded me that `docs` is a first-class citizen of the ecosystem!

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+for their OSS products. We would never go that route, but other businesses have
+gone the copy-left route (Elastic, Grafana, etc) for similar reasons--to avoid
+business risk with large competitors, especially the hyper-scalers (AWS, GCP,
+Azure, etc).
+
+In our business model we have a distinct advantage because of our ability to
+delivery better OSS value than nearly any other defense contractor and better
+OSS financial outlook than most OSS companies due to our trust we have built
+with various government organizations. However, in some ways, the DoD Primes are
+our hyper-scaler equivalents, they are in the same markets with considerable
```
Comment (Racer159) [inline docs/adrs/0002-apache-2.0-for-all-uds-products.md:23]:
```suggestion
our hyper-scaler equivalents, they are in the same markets with considerably
```

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+more resources and deeper relationships. This means our moat, when looking at
+other successful defense contractors in particular, is really around out ability
```
Comment (Racer159) [inline docs/adrs/0002-apache-2.0-for-all-uds-products.md:25]:
```suggestion
other successful defense contractors in particular, is really around our ability
```

File: docs/adrs/0002-apache-2.0-for-all-uds-products.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+to
+
+1. Create compelling FOSS technology
+2. Bring in unique talent not typically found in the defense industrial base
+3. Grow an incredibly talented delivery team that can leverage our FOSS
+   technology to move faster than others.
+
+Copy-left is still Open Source, but it does cause many organizations to be
+hesitant to adopt. Because we believe so deeply in wide adoption of FOSS and
+mission impact is still what drives us, we are choosing to keeping everything
```
Comment (Racer159) [inline docs/adrs/0002-apache-2.0-for-all-uds-products.md:35]:
```suggestion
mission impact is still what drives us, we are choosing to keep everything
```

**Review by Racer159 — CHANGES_REQUESTED** [review]
(decided to take a second look)

**Comment by daveworth** [comment]
updated! Thanks @Racer159

**Review by Racer159 — APPROVED** [review]

**Pull request merged**